source/index.tsx

```diff
@@ -71,7 +74,12 @@ const ExpandingTextarea: FC<TextareaProps> = ({
   forwardedRef,
   ...props
 }: TextareaProps) => {
-  const ref = forwardedRef || createRef<HTMLTextAreaElement>()
+  const isForwardedRefFn = typeof forwardedRef === 'function'
+  const ref = (
+    isForwardedRefFn || !forwardedRef
+      ? createRef<HTMLTextAreaElement>()
```

Comment:
Curious on the choice of `createRef` as won't this create a new `ref` per render, as opposed to persisting the same one across renders with `useRef`?

The conditional logic is a lot harder with `useRef` though, because of the nature of hooks

Reply:
This is a good point. It's easy enough with `useRef` -- standby. Good :eyes:

Reply:
Updated in https://github.com/rpearce/react-expanding-textarea/pull/55/commits/b6143aa50a2aa93a42a992b9b609bfe141dcfbae

Reply:
Nice change 👍